aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,125 @@
+
+MVC uses data annotations in addition to type metadata from model properties to validate data and display any error messages. It does so by rendering `HTML5 ``data-`` attributes  <http://w3c.github.io/html/dom.html#embedding-custom-non-visible-data-with-the-data-attributes>`_ in the form elements that need validation, as shown below. MVC generates the ``data-`` attributes for both built-in and custom attributes. The ``data-val-required`` attribute below contains an error message to display if the user doesn't fill in the release date field, and that message displays in the accompanying ``<span>`` element.
+
+.. code-block:: html
+ :emphasize-lines: 3
+		
+  <input name="ReleaseDate" class="form-control" id="ReleaseDate" 
+  type="datetime" value="" 
+  data-val-required="The ReleaseDate field is required." 
+  data-val="true">
+        
+  <span class="text-danger field-validation-valid" 
+  data-valmsg-replace="true" 
+  data-valmsg-for="ReleaseDate">
+            
+After the form fields pass client side validation, the ``ModelStateDictionary`` is populated with values. When the HTTP request reaches MVC it invokes the appropriate action method for processing.
+
+Remote valiation
+----------------
+Remote validation is a great feature to use when you need to validate data on the client against data on the server. For example, your app may need to verify whether an email or user name is already in use, and it must query a large amount of data to do so. Downloading large sets of data for validating one field consumes too many resources to be worthwile. An alternative is to make a round-trip request to validate a field; however, doing so diminishes the user's experience, especially when other fields quickly pass client side validation. The solution is to use remote validation that will make the UX better with instant validation and high performance.
+
+You can implement remote validation in a two step process. First, you must annotate your model with the ``[Remote]`` attribute. The ``[Remote]`` attribute accepts multiple overloads you can use to direct client side JavaScript to the appropriate code to call. The example points to the ``VerifyEmail`` action method of the ``Users`` controller. 
+
+.. literalinclude:: validation/sample/User.cs
+ :language: c#
+ :lines: 9-17
+ :dedent: 4
+ 
+The second step is putting the validation code in the corresponding action method. In this case it's the ``VerifyEmail`` method in the ``UsersController``. It returns a ``JsonResult`` that the client side can use to proceed or pause and display a validation error if needed.
+ 
+.. literalinclude:: validation/sample/UsersController.cs
+ :language: c#
+ :lines: 9-17   
+ :dedent: 4
+
+Now when users enter an email, JavaScript in the view makes a remote call to see if that email has been taken, and if so, then displays the error message. Otherwise, the user can submit the form as usual.  
```

Comment:
Needs another section on `IClientModelValidator`. `[Remote]` is an _example_ of the MVC extensibility for client-side validation. Attributes that implement this `interface` and are applied to model types or properties can add HTML attributes to generated fields.